csrc/fusion.cpp

```diff
@@ -320,7 +320,11 @@ void Fusion::replaceOutput(Val* output, Val* replacement) {
     }
     if (output->getValType().value() == ValType::TensorView) {
       output->setIsFusionOutput(false);
-      output->as<TensorView>()->setMemoryType(MemoryType::Local);
+      // If `output` is both an input and an output before the replacement,
+      // don't localize it.
+      if (!output->isFusionInput()) {
+        output->as<TensorView>()->setMemoryType(MemoryType::Local);
```

Comment:
Just for extra safety, could you also add an assertion in `setMemoryType` that tensors must not be an input nor an output?

Reply:
The assertion already exists (https://github.com/NVIDIA/Fuser/blob/6c6f3a40e09e6f8bece80b9b79c543945846c71b/csrc/tensor_view.cpp#L1227-L1231) and was indeed triggered when I worked on https://github.com/NVIDIA/Fuser/pull/2529. 

The assertion says "if I/O, then assert global", which is equivalent to "if not global, then assert not I/O".